pages/blog/_posts/2020-10-01-what-is-short-messages.md

```diff
@@ -0,0 +1,214 @@
+purposes; you might not want other networks and hackers to know about
+who you are.
+
+For *MT* requests there is also a flag which is called
+`moreMessagesToSend` which is used if (god forbid) you would break the
+protocol and send a text message longer than 115 bytes. The message is
+then breaked up into chunks which starts with an empty *TCAP* `Begin`
+message followed with the actual text message in `Continue`
+messages. In the end (_hehe_) the `End` message is transmitted as a
+response and the transaction is finished.
+
+When one of these chunk (or the full message if lesser than 115 bytes)
+is delivered to the recipient, a response is sent back with the
+`MT-Forward-SM` containing a delivery status.
+
+OR, in the case that you are sending an *SMS* to a recipient in another
+network and the recipient is not available, it might contain the time
+the receiving *SMSC* will hold the message and retry to send it to the
+recipient.
+
+_At least this is main idea I think..._
+
+# Differences in MAP versions for SMS
+
+The `moreMessagesToSend` flag was implemented for version 2, so it exist
+only after version 1.
+
+In version 1, the dialogue portion was not invented and all chunks are
+sent in new *TCAP* dialogues. The size of the user data could then
+be 140 bytes.
+
+In version 1 and version 2 there is no difference between *MT* and
+*MO*. Everything is sent as another type of message `Forward-SM`,
+which does not include any privacy correlation ids, and there are no
+fancy responses with delivery status. There is still an
+acknowledgement, but is in a form of an empty message.
+
+Only way to see difference between an MO and a MT in version 1 is to
+look at the addresses and see if they are either coming from an SMSC
+or going to an SMSC.
+
+Ok, to recap, what do we have now
+
+- `Begin`, `Continue`, `End`, `Abort` messages.
+- Dialogue handshake in the first request/response messages sent.
+- `MT-Forward-SM`, `MO-Forward-SM`, `Forward-SM`
+- Mobile phones and *SMSCs*
+
+_Let's talk a bit about the last item_
+
+# Different SMSCs
+
+So you are either on your home network or on a different (roaming)
+network. If you are not on your operators network you are a visitor
+and thus roaming. The same goes for the recipient of the *SMS*.
+
+So the mobile originating message is sent to the *SMSC* in the network
+you are in. That *SMSC* then sends a mobile terminating message to the
+*SMSC* in the recipients network, which in turn sends a mobile
+terminating message to the recipient.
+
+# What about 4G/LTE networks?
+
+_Ouch._
+
+*LTE* networks does not use any of the *M3UA*, *SCCP*, *TCAP*, *MAP*
+protocols. In *LTE* networks the main message type is Diameter which
+doesn't contain fragmentation and can contain larger
+messages. Everything is sent in one request and every request is
+answered with a response.
+
+To make *SMSes* work on *LTE* networks a new node was invented which
+translates Diameter messages to *SIGTRAN* messages. This node sits in
```

Comment:
In most cases, SMS are simply sent as always via SS7 to the MSC, which then uses the SGs interface to forward it to the MME, which delivers it to the UE